crates/signal/src/lib.rs

```diff
@@ -76,6 +76,11 @@ mod bootstrap;
 pub use bootstrap::{SignalConfig, SignalSvc};
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+/// The kind of signal to listen for.
+/// The `Interrupt` maps to `SIGINT` on Unix, and `Ctrl-C` on Windows.
+/// The `Terminate` maps to `SIGTERM` on Unix, and `Ctrl-Close` on Windows.
+/// The `Unix` and `Windows` are plataform specific signals. `Unix` maps to
+/// `UnixSignalKind`, and `Windows` maps to `WindowsSignalKind`.
```

Comment:
This comment is on the struct, we should just say something like
```
The type of signal to listen for. 
``` 

and then we should put the stuff about `Interrupt` above the variant.

Comments should also go above the derive statements.